extensions/wikia/Search/classes/TopWikiArticles.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Wikia\Search;
+
+class TopWikiArticles {
+	/**
+	 * Used for changing mem cache key for top articles snippet (after changed will end up in purging cache)
+	 */
+	const TOP_ARTICLES_CACHE = 1;
+
+	/**
+	 * 5 days, one business week
+	 */
+	const CACHE_TTL = 86400 * 5;
+
+	/**
+	 * We want to return articles with thumbnails, but not all articles
+	 * returned from ArticlesApiController:getTop() have them, that is why we ask for
+	 * more articles than we will return
+	 */
+	const GET_NUMBER_OF_TOP_ARTICLES_FROM_DB = 12;
+	const TOP_ARTICLES_RESULT_LIMIT = 5;
+
+	static public function getArticlesWithCache( $cityId, $isGridLayoutEnabled ) {
+		$cacheKey = wfMemcKey(
+			__CLASS__,
+			'WikiaSearch',
+			'topWikiArticles',
+			$cityId,
+			static::TOP_ARTICLES_CACHE,
+			$isGridLayoutEnabled
+		);
+
+		return \WikiaDataAccess::cache(
+			$cacheKey,
+			86400 * 5, // 5 days, one business week
+			function () {
+				return static::getArticles();
+			}
+		);
+	}
+
+	static public function getArticles() {
+		global $wgLang;
+
+		$pages = [];
+
+		try {
+			$app = \F::app();
+			$pageData = $app
+				->sendRequest( 'ArticlesApiController', 'getTop', [ 'namespaces' => 0, 'limit' => static::GET_NUMBER_OF_TOP_ARTICLES_FROM_DB ] )
+				->getData();
+			$ids = [];
+
+			foreach ( $pageData['items'] as $item ) {
+				$ids[] = $item['id'];
+			}
+
+			$params = [ 'ids' => implode( ',', $ids ), 'height' => 90, 'width' => 90, 'abstract' => 120 ];
+			$detailResponse = $app->sendRequest( 'ArticlesApiController', 'getDetails', $params )->getData();
+
+			foreach ( $detailResponse['items'] as $item ) {
```

Comment:
Do we need this for loop?
it looks that code here is related to presentation layer.